### PR TITLE
Issue pickle loss of parent properties 

### PR DESCRIPTION
**Pull request by HFragnaud**
fix for Epoch, Event, IrregularlySampledEvent and tests associated

### CODE REVIEW

**Comment by HFragnaud** [comment]
@apdavison could you review this please?

File: neo/core/epoch.py [hunk 1]
```diff
@@ -19,13 +19,15 @@
...
 def _new_epoch(cls, times=None, durations=None, labels=None, units=None,
-                name=None, description=None, file_origin=None, annotations = None):
+                name=None, description=None, file_origin=None, annotations = None, segment=None):
     '''
-    A function to map Event.__new__ to function that
+    A function to map epoch.__new__ to function that
     does not do the unit checking. This is needed for pickle to work. 
     '''
-    return Epoch( times=times, durations=durations, labels=labels, units=units, name=name, file_origin=file_origin,
+    e=Epoch( times=times, durations=durations, labels=labels, units=units, name=name, file_origin=file_origin,
```
Comment (apdavison) [inline neo/core/epoch.py]: Please use spaces around "=" (except for function keyword arguments) (https://www.python.org/dev/peps/pep-0008/#other-recommendations)

File: neo/core/event.py [hunk 2]
```diff
@@ -20,13 +20,15 @@
...
 def _new_event(cls, signal, times = None, labels=None, units=None, name=None, 
                file_origin=None, description=None,
-               annotations=None):
+               annotations=None, segment=None):
     '''
     A function to map Event.__new__ to function that
     does not do the unit checking. This is needed for pickle to work. 
     '''
-    return Event(signal=signal, times=times, labels=labels, units=units, name=name, file_origin=file_origin,
+    e=Event(signal=signal, times=times, labels=labels, units=units, name=name, file_origin=file_origin,
```
Comment (apdavison) [inline neo/core/event.py]: spaces around "="

**Comment by apdavison** [comment]
Looks great, just a couple of PEP8 formatting issues to fix then it can be merged.

**Pull request merged**